Explorer/View/DatePickerView.swift

```diff
@@ -17,36 +17,39 @@ enum DatePickerMode {
 class DatePickerView: UIView {
     
     let picker = UIDatePicker()
-    let doneButton: UIButton = {
-        let button = UIButton()
-        button.setTitle("Done", for: .normal)
-        button.setTitleColor(.blue, for: .normal)
-        return button
+    
+    let toolbar: UIToolbar = {
+        let bar = UIToolbar()
+        let flexspace = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        let doneButton = UIBarButtonItem(title: "Done", style: .done, target: self, action: #selector(CreateLogViewController.setDate))
```

Comment:
Last thing: you do want to set this action inside `CreateLogViewController`. Views shouldn’t really know about their parents - the date picker shouldn’t even know `CreateLogViewController` exists. Move this `addTarget` call to `CreateLogViewController`.